internal/pkg/cleaner/cleaner.go

```diff
@@ -0,0 +1,144 @@
+package cleaner
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync/atomic"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/zajca/go-executor/internal/pkg/client"
+	"github.com/zajca/go-executor/internal/pkg/job"
+	"golang.org/x/sync/syncmap"
+)
+
+type JobCleaner struct {
+	broadcaster *client.Broadcaster
+	jobsToClean syncmap.Map
+	jobsCounter uint32
+}
+
+func NewJobCleaner(b *client.Broadcaster) *JobCleaner {
+	return &JobCleaner{
+		broadcaster: b,
+		jobsToClean: syncmap.Map{},
+	}
+}
+
+func (c *JobCleaner) HandleJob(j *job.Job, l echo.Logger) {
+	l.Info(fmt.Sprintf("Job '%s' will be cleared.", j.JobId))
+	if err := os.Remove(j.Path.PID); err != nil {
+		l.Error(err)
+	}
+	c.jobsToClean.Store(j, false)
+	atomic.AddUint32(&c.jobsCounter, 1)
+}
+
+func (c *JobCleaner) Count() uint32 {
+	return c.jobsCounter
+}
+
+func (c *JobCleaner) Run(jc chan<- *job.Job, l echo.Logger) {
+	for {
+		if !c.broadcaster.HasAnyClient() {
+			if c.jobsCounter != 0 {
+				l.Info(fmt.Sprintf("'%d' jobs to clean, but no client is connected.", c.jobsCounter))
+			}
+			time.Sleep(1 * time.Second)
+			continue
+		}
+
+		if c.jobsCounter == 0 {
+			l.Info("No jobs to clean.")
+			time.Sleep(1 * time.Second)
```

Comment:
Tu by si mohol tiez pouzit `ticker`, takto to nebude presne, vzdy sa to spomali o pracu co sa vykonala.